src/main/java/com/revature/services/QuestionLogicImpl.java

```diff
@@ -18,11 +21,62 @@
 	
 	@Autowired
 	private QuestionRepository dao;
+	
+	@Autowired
+	private QuestionEvalRepository qEvalDao;
+	
+	@Autowired
+	private QuestionCommentRepository commentDao;
+	
+	@Autowired
+	private SubjectRepository subjectDao;
+	
+//CREATE-----------------------------------
+	@Override
+	@Transactional
+	public QuestionPool createQuestion(QuestionPool question) {
+		
+		if (question.getMaxCommunicationScore() == null) {
+			throw new ConstraintViolationException("Missing required field maxCommunicationScore (Integer)", null);
+		}
+		
+		if (question.getMaxKnowledgeScore() == null) {
+			throw new ConstraintViolationException("Missing required field maxKnowledgeScore (Integer)", null);
+		}
+		
+		if (question.getQuestionText() == null) {
+			throw new ConstraintViolationException("Missing required field questionText (String)", null);
+		}
+		
+		if (question.getSubject() == null || question.getSubject().getId() == null) {
+			throw new ConstraintViolationException("Missing required field subject.id (Integer)", null);
+		}
+		
+		question.setUseCount(0);
+		question = dao.save(question);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Introduce a new variable instead of reusing the parameter "question". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.revaturelabs.com/coding_rules#rule_key=squid%3AS1226)